images/dvcr/werf.inc.yaml

```
@@ -1,15 +1,39 @@
 ---
-image: {{ $.ImageName }}-original-registry2
-final: false
-from: {{ .Images.BASE_CONTAINER_REGISTRY }}
----
 image: {{ $.ImageName }}
 from: {{ .Images.BASE_SCRATCH }}
 import:
-- image: {{ $.ImageName }}-original-registry2
-  add: /bin
+- image: {{ $.ImageName }}-builder
+  add: /container-registry-binary
   to: /bin
   after: install
-  includePaths:
-  - registry
-# Registry configuration is stored in configmap: templates/dvcr/configmap.yaml 
\ No newline at end of file
+# Registry configuration is stored in configmap: templates/dvcr/configmap.yaml 
+---
+{{- $version := "2.8.3" }}
+
+image: {{ $.ImageName }}-builder
+final: false
+fromImage: base-alt-p10
+shell:
+  install:
+    - |
+      apt-get update && apt-get install -y \
+      git openssh golang
+
+      export GOPATH=$(go env GOPATH)
+      export GOROOT=$(go env GOROOT)
+      export PATH=$PATH:$GOPATH/bin:$GOROOT/bin
+      export GO111MODULE=off
+      
+      mkdir -p /container-registry-binary
+      mkdir -p $GOPATH/src/github.com/docker
+
+      cd $GOPATH/src/github.com/docker
+      git clone --depth 1 --branch v{{ $version }} {{ $.SOURCE_REPO }}/docker/distribution.git
+      cd distribution
+
+      go build -o /container-registry-binary/ -gcflags "-N -l" -ldflags '-s -w -X registry/version.Version=v{{ $version }} -X registry/version.Revision=v{{ $version }} -X registry/version.Package=v{{ $version }}' ./cmd/...
```

```suggestion
      go build -o /container-registry-binary/ -gcflags "-N -l" -ldflags '-s -w -X registry/version.Version=v{{ $version }} -X registry/version.Revision=v{{ $version }}' ./cmd/registry
```

1. Default version.Package (github.com/docker/distribution) is ok.
2. No need to build other binaries, we need only registry binary.